KISS crashes upon some search strings

Possibly in conjunction with work profile but no sure, can't quite see a pattern.
![Screenshot_20201031-164929_Market Feedback Agent](https://user-images.githubusercontent.com/41674213/97808679-8b0d1900-1c68-11eb-824b-5f60b34b8250.jpg)
![Screenshot_20201031-164938_Market Feedback Agent](https://user-images.githubusercontent.com/41674213/97808682-8e080980-1c68-11eb-878b-1f85105d47d1.jpg)
![Screenshot_20201031-164950_Market Feedback Agent](https://user-images.githubusercontent.com/41674213/97808685-8f393680-1c68-11eb-8cc7-1748ad20e9e9.jpg)
![Screenshot_20201031-165001_Market Feedback Agent](https://user-images.githubusercontent.com/41674213/97808688-92342700-1c68-11eb-9cfe-9cb18099e054.jpg)

I've crashed it, generated a bug report and found this stacktrace, hope it can help:

```11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: FATAL EXCEPTION: main
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: Process: fr.neamar.kiss, PID: 14076
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: java.lang.IllegalArgumentException: To query by shortcut ID, package name must also be set
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.os.Parcel.createException(Parcel.java:2075)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:2039)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:1987)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.content.pm.ILauncherApps$Stub$Proxy.getShortcuts(ILauncherApps.java:1271)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.content.pm.LauncherApps.getShortcuts(LauncherApps.java:888)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at fr.neamar.kiss.result.ShortcutsResult.getDrawable(ShortcutsResult.java:12)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at fr.neamar.kiss.result.ShortcutsResult.display(ShortcutsResult.java:33)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at fr.neamar.kiss.adapter.RecordAdapter.getView(RecordAdapter.java:1)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.widget.AbsListView.obtainView(AbsListView.java:2412)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.widget.ListView.makeAndAddView(ListView.java:2071)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.widget.ListView.fillUp(ListView.java:832)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.widget.ListView.layoutChildren(ListView.java:1809)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.widget.AbsListView.onLayout(AbsListView.java:2202)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.view.View.layout(View.java:21973)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.view.ViewGroup.layout(ViewGroup.java:6263)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.view.View.layout(View.java:21973)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.view.ViewGroup.layout(ViewGroup.java:6263)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.widget.RelativeLayout.onLayout(RelativeLayout.java:1103)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.view.View.layout(View.java:21973)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.view.ViewGroup.layout(ViewGroup.java:6263)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.view.View.layout(View.java:21973)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.view.ViewGroup.layout(ViewGroup.java:6263)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1829)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1673)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.widget.LinearLayout.onLayout(LinearLayout.java:1582)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.view.View.layout(View.java:21973)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.view.ViewGroup.layout(ViewGroup.java:6263)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at com.android.internal.policy.DecorView.onLayout(DecorView.java:799)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.view.View.layout(View.java:21973)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.view.ViewGroup.layout(ViewGroup.java:6263)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:3108)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2618)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1749)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7692)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:966)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.view.Choreographer.doCallbacks(Choreographer.java:790)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.view.Choreographer.doFrame(Choreographer.java:725)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:951)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:883)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:100)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:241)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7604)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:941)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: Caused by: android.os.RemoteException: Remote stack trace:
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at com.android.server.pm.LauncherAppsService$LauncherAppsImpl.getShortcuts(LauncherAppsService.java:664)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.content.pm.ILauncherApps$Stub.onTransact(ILauncherApps.java:589)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.os.Binder.execTransactInternal(Binder.java:1036)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 	at android.os.Binder.execTransact(Binder.java:1003)
11-20 16:35:09.218 10256 14076 14076 E AndroidRuntime: 
11-20 16:35:09.220  1000  2072 14117 I DropBoxManagerService: add tag=data_app_crash isTagEnabled=true flags=0x2
11-20 16:35:09.220  1000  2072  4184 W ActivityTaskManager:   Force finishing activity fr.neamar.kiss/.MainActivity
```

--- a/app/src/main/java/fr/neamar/kiss/result/ShortcutsResult.java
+++ b/app/src/main/java/fr/neamar/kiss/result/ShortcutsResult.java
@@ -17,6 +17,7 @@ import android.os.Build;
 import android.os.UserHandle;
 import android.os.UserManager;
 import android.preference.PreferenceManager;
+import android.text.TextUtils;
 import android.view.View;
 import android.view.ViewGroup;
 import android.view.WindowManager;
@@ -146,7 +147,7 @@ public class ShortcutsResult extends Result {
             UserManager userManager = (UserManager) context.getSystemService(Context.USER_SERVICE);
             assert launcherApps != null;
 
-            if (launcherApps.hasShortcutHostPermission()) {
+            if (launcherApps.hasShortcutHostPermission() && !TextUtils.isEmpty(shortcutPojo.packageName)) {
                 LauncherApps.ShortcutQuery query = new LauncherApps.ShortcutQuery();
                 query.setPackage(shortcutPojo.packageName);
                 query.setShortcutIds(Collections.singletonList(shortcutPojo.getOreoId()));
@@ -203,20 +204,22 @@ public class ShortcutsResult extends Result {
             return;
         }
 
-        LauncherApps.ShortcutQuery query = new LauncherApps.ShortcutQuery();
-        query.setPackage(shortcutPojo.packageName);
-        query.setShortcutIds(Collections.singletonList(shortcutPojo.getOreoId()));
-        query.setQueryFlags(FLAG_MATCH_DYNAMIC | FLAG_MATCH_MANIFEST | FLAG_MATCH_PINNED);
+        if (!TextUtils.isEmpty(shortcutPojo.packageName)) {
+            LauncherApps.ShortcutQuery query = new LauncherApps.ShortcutQuery();
+            query.setPackage(shortcutPojo.packageName);
+            query.setShortcutIds(Collections.singletonList(shortcutPojo.getOreoId()));
+            query.setQueryFlags(FLAG_MATCH_DYNAMIC | FLAG_MATCH_MANIFEST | FLAG_MATCH_PINNED);
 
-        List<UserHandle> userHandles = launcherApps.getProfiles();
+            List<UserHandle> userHandles = launcherApps.getProfiles();
 
-        // Find the correct UserHandle, and launch the shortcut.
-        for (UserHandle userHandle : userHandles) {
-            if (userManager.isUserRunning(userHandle)) {
-                List<ShortcutInfo> shortcuts = launcherApps.getShortcuts(query, userHandle);
-                if (shortcuts != null && shortcuts.size() > 0 && shortcuts.get(0).isEnabled()) {
-                    launcherApps.startShortcut(shortcuts.get(0), v.getClipBounds(), null);
-                    return;
+            // Find the correct UserHandle, and launch the shortcut.
+            for (UserHandle userHandle : userHandles) {
+                if (userManager.isUserRunning(userHandle)) {
+                    List<ShortcutInfo> shortcuts = launcherApps.getShortcuts(query, userHandle);
+                    if (shortcuts != null && shortcuts.size() > 0 && shortcuts.get(0).isEnabled()) {
+                        launcherApps.startShortcut(shortcuts.get(0), v.getClipBounds(), null);
+                        return;
+                    }
                 }
             }
         }

